src/gt4py/cartesian/definitions.py

```diff
@@ -131,16 +131,16 @@ class GTError(Exception):
 
 
 class GTSyntaxError(GTError):
-    def __init__(self, message, *, frontend):
+    def __init__(self, message: str, *, frontend: str):
```

Comment:
Just a question: did you forget to add the `None` return type hint or you did it on purpose?  I know it's not longer needed so I'm just curious.